r/aws_eks_identity_provider_config: Acceptance tests fail with creation timeout

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

The `aws_eks_identity_provider_config` resource acceptance tests are failing with errors similar to

```
=== RUN   TestAccAWSEksIdentityProviderConfig_Tags
=== PAUSE TestAccAWSEksIdentityProviderConfig_Tags
=== CONT  TestAccAWSEksIdentityProviderConfig_Tags
resource_aws_eks_identity_provider_config_test.go:203: Step 1/4 error: Error running apply: exit status 1
Error: error waiting for EKS Identity Provider Config (tf-acc-test-8280502334800591763:tf-acc-test-8280502334800591763) association: context deadline exceeded
on terraform_plugin_test.tf line 68, in resource "aws_eks_identity_provider_config" "test":
68: resource "aws_eks_identity_provider_config" "test" {
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: error deleting EKS Cluster (tf-acc-test-8280502334800591763): ResourceInUseException: Cannot delete because cluster tf-acc-test-8280502334800591763 currently has update 197f1d1f-407a-4367-ba7b-a6aeb95b9d62 in progress
{
RespMetadata: {
StatusCode: 409,
RequestID: "ea809498-a444-4aba-8d3b-4834899a27b2"
},
ClusterName: "tf-acc-test-8280502334800591763",
Message_: "Cannot delete because cluster tf-acc-test-8280502334800591763 currently has update 197f1d1f-407a-4367-ba7b-a6aeb95b9d62 in progress"
}
--- FAIL: TestAccAWSEksIdentityProviderConfig_Tags (2236.12s)
```

--- a/.changelog/20561.txt
+++ b/.changelog/20561.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_eks_identity_provider_config: Increase Create and Delete timeouts to 40 minutes
+```

--- a/aws/resource_aws_eks_identity_provider_config.go
+++ b/aws/resource_aws_eks_identity_provider_config.go
@@ -22,10 +22,10 @@ import (
 
 func resourceAwsEksIdentityProviderConfig() *schema.Resource {
 	return &schema.Resource{
-		CreateContext: resourceAwsEksIdentityProviderConfigCreate,
-		ReadContext:   resourceAwsEksIdentityProviderConfigRead,
-		UpdateContext: resourceAwsEksIdentityProviderConfigUpdate,
-		DeleteContext: resourceAwsEksIdentityProviderConfigDelete,
+		CreateWithoutTimeout: resourceAwsEksIdentityProviderConfigCreate,
+		ReadWithoutTimeout:   resourceAwsEksIdentityProviderConfigRead,
+		UpdateWithoutTimeout: resourceAwsEksIdentityProviderConfigUpdate,
+		DeleteWithoutTimeout: resourceAwsEksIdentityProviderConfigDelete,
 
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
@@ -34,8 +34,8 @@ func resourceAwsEksIdentityProviderConfig() *schema.Resource {
 		CustomizeDiff: SetTagsDiff,
 
 		Timeouts: &schema.ResourceTimeout{
-			Create: schema.DefaultTimeout(25 * time.Minute),
-			Delete: schema.DefaultTimeout(25 * time.Minute),
+			Create: schema.DefaultTimeout(40 * time.Minute),
+			Delete: schema.DefaultTimeout(40 * time.Minute),
 		},
 
 		Schema: map[string]*schema.Schema{

--- a/website/docs/r/eks_identity_provider_config.html.markdown
+++ b/website/docs/r/eks_identity_provider_config.html.markdown
@@ -56,8 +56,8 @@ In addition to all arguments above, the following attributes are exported:
 
 `aws_eks_identity_provider_config` provides the following [Timeouts](https://www.terraform.io/docs/configuration/blocks/resources/syntax.html#operation-timeouts) configuration options:
 
-* `create` - (Default `25 minutes`) How long to wait for the EKS Identity Provider Configuration to be associated.
-* `delete` - (Default `25 minutes`) How long to wait for the EKS Identity Provider Configuration to be disassociated.
+* `create` - (Default `40 minutes`) How long to wait for the EKS Identity Provider Configuration to be associated.
+* `delete` - (Default `40 minutes`) How long to wait for the EKS Identity Provider Configuration to be disassociated.
 
 ## Import
 

